homeassistant/components/evohome/__init__.py

```diff
@@ -118,91 +113,154 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Create a (EMEA/EU-based) Honeywell TCC system."""
+class EvoAuthenticator:
+    """Class for evohome client instantiation & authentication."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the evohome broker and its data structure."""
+
+        self.hass = hass
+
+        self._session = async_get_clientsession(hass)
+        self._store = Store[dict[str, Any]](hass, STORAGE_VER, STORAGE_KEY)
+
+        # the main client, which uses the newer API
+        self.client_v2: evo.EvohomeClient | None = None
+        self._tokens: dict[str, Any] = {}
+
+        # the older client can be used to obtain high-precision temps (only)
+        self.client_v1: ev1.EvohomeClient | None = None
+        self.session_id: str | None = None
+
+    async def authenticate(self, username: str, password: str) -> bool:
+        """Check the user credentials against the web API."""
+
+        if self.client_v2 is None:
+            await self._load_auth_tokens(username)
+
+            client_v2 = evo.EvohomeClient(
+                username,
+                password,
+                **self._tokens,
+                session=self._session,
+            )
+
+        else:  # force a re-authentication
+            client_v2 = self.client_v2
+            client_v2._user_account = {}  # noqa: SLF001
+
+        try:
+            await client_v2.login()
+        except evo.AuthenticationFailed as err:
```

Comment:
Skip catching this here, catch that in async_setup instead. In config flow later you want to raise another exception.